include/aktualizr-lite/tuf/aklite-tuf.h

```diff
@@ -0,0 +1,248 @@
+  }
+
+  virtual Json::Value fetchSnapshot() {
+    return fetchRole(Uptane::Role::Snapshot(), Uptane::kMaxSnapshotSize, Uptane::Version());
+  }
+
+  virtual Json::Value fetchTargets() {
+    return fetchRole(Uptane::Role::Targets(), Uptane::kMaxImageTargetsSize, Uptane::Version());
+  }
+};
+
+// TufRepoSource implementation for fetching local meta-dat
+class AkLocalTufRepoSource : public TufRepoSource {
+ private:
+  boost::filesystem::path src_dir_;
+  virtual Json::Value fetchFile(boost::filesystem::path meta_file_path) {
+    std::cout << "fetchFile " << meta_file_path << std::endl;
+    if (!boost::filesystem::exists(meta_file_path)) throw NotFoundException(meta_file_path.string());
+
+    return Utils::parseJSONFile(meta_file_path);
+  }
+
+ public:
+  class NotFoundException : public std::runtime_error {
+   public:
+    NotFoundException(const std::string& path) : std::runtime_error("Metadata hasn't been found; file path: " + path) {}
+  };
+
+  AkLocalTufRepoSource(std::vector<boost::filesystem::path> cfg_dirs, boost::filesystem::path src_dir_in) {
+    src_dir_ = src_dir_in;
+  }
+
+  // DISCUSS: limit to max version?
+  virtual Json::Value fetchRoot(int version) { return fetchFile(src_dir_ / (std::to_string(version) + ".root.json")); }
+
+  virtual Json::Value fetchTimestamp() { return fetchFile(src_dir_ / "timestamp.json"); }
+
+  virtual Json::Value fetchSnapshot() { return fetchFile(src_dir_ / "snapshot.json"); }
+
+  virtual Json::Value fetchTargets() { return fetchFile(src_dir_ / "targets.json"); }
+};
+
+// Wrapper around any TufRepoSource implementation to make it usable directly by libaktualizr,
+// by implementing Uptane::IMetadataFetcher interface
+class FetcherWrapper : public Uptane::IMetadataFetcher {
+ private:
+  std::shared_ptr<TufRepoSource> repo_src;
+
+ public:
+  FetcherWrapper(std::shared_ptr<TufRepoSource> src) { repo_src = src; }
+  void fetchRole(std::string* result, int64_t maxsize, Uptane::RepositoryType repo, const Uptane::Role& role,
+                 Uptane::Version version) const override {
+    Json::Value json;
+    if (role == Uptane::Role::Root())
+      json = repo_src->fetchRoot(version.version());
+    else if (role == Uptane::Role::Timestamp())
+      json = repo_src->fetchTimestamp();
+    else if (role == Uptane::Role::Snapshot())
+      json = repo_src->fetchSnapshot();
+    else if (role == Uptane::Role::Targets())
+      json = repo_src->fetchTargets();
+    else
+      throw std::runtime_error("Invalid TUF Role " + role.ToString());
+
+    if (json) {
+      Json::StreamWriterBuilder builder;
+      builder["indentation"] = "";
+      *result = Json::writeString(builder, json);
+    }
+  }
+
+  void fetchLatestRole(std::string* result, int64_t maxsize, Uptane::RepositoryType repo,
+                       const Uptane::Role& role) const override {
+    fetchRole(result, maxsize, repo, role, Uptane::Version());
+  }
+};
+
+// DISCUSS: What fields do we need? Explicit fields or map?
```

Comment:
I would make it `struct` and extends it (add fields) based on needs. Just add only those fields that are needed by the current client(s).